DirectProgramming/C++SYCL_FPGA/Tutorials/Features/hls_flow_interfaces/invocation_interfaces/README.md

```diff
@@ -190,6 +190,10 @@ for (int i = 0; i < count; i++) {
 q.wait();
 ```
 
+> **Note**: As per the SYCL language semantics, separate invocations of a kernel are **independent**. This means that you can't make assumptions about memory ordering or memory dependences between kernel invocations. Make sure you use synchronization mechanisms such as the `.wait()` function or *atomic operations* to avoid race conditions. \
+If you want to guarantee sequential equivalence, you can also write your kernel with a `while(1)` loop in the kernel body instead of using a pipelined kernel. \
+In particular, a repeatedly-invoked kernel that would have worked correctly in legacy HLS may result in undefined behavior in SYCL and may not function as you expect. 
```

Comment:
I'm not sure legacy HLS should be mentioned here, @whitepau any thoughts?

Reply:
I don't have a strong feeling one way or another, we can re-word to 

In particular, a repeatedly-invoked kernel *with memory dependences* ~that would have worked correctly in legacy HLS~ will result in undefined behavior in SYCL and may not function as you expect. 

i don't like to say something vague like 'a repeatedly-invoked kernel will result in undefined behaviour' because that sounds like the whole feature doesn't work.

Reply:
Sounds good to me